### ISSUE TITLE
[FR] Support for Azure US Gov Cloud & Azurite Storage URIs in AzureBlobArtifactRepository

### ISSUE THREAD

**Issue by vladmsft** [mlflow/mlflow#9657]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

Currently the utility only supports parsing China Cloud or Commercial Cloud Azure Blob Storage URI's - this would modify parsing of Azure Blob Storage URI to include a path pointed to Azure US Gov Cloud or a Azurite Storage Emulator

### Motivation

> #### What is the use case for this feature?
Interaction with Blob Storage on Azure US Gov Cloud or a local storage emulator

> #### Why is this use case valuable to support for MLflow users in general?
Opens artifact implementation for another cloud option as well as local emulation

> #### Why is this use case valuable to support for your project(s) or organization?
Used for downloading blob data from aforementioned cloud and emulator when using the MLFlow library

> #### Why is it currently difficult to achieve this use case?
Custom code is necessary to achieve above when using MLFlow 

### Details

Modification to the `parse_wasbs_uri` method inside of https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/azure_blob_artifact_repo.py would include additional RegEx matching on a `usgovcloudapi.net` and `azurite` suffixes. 

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
@vladmsft this sounds great! We will look forward to your PR!

**Comment by templierw** [comment]
@vladmsft almost 2 years, any progress? Still interested in this issue?

**Comment by ahringer** [comment]
> [@vladmsft](https://github.com/vladmsft) almost 2 years, any progress? Still interested in this issue?

Yes still interest. I have a workaround patch for our installation, but I'm sure others must be hitting this issue.

**Comment by chcharlet** [comment]
Hi, this really is a bit of a pain, specifically for running mockups in CI. I am with @ahringer here, in that I run a monkeypatched version in our CI, but obviously that's not a great solution.

I'd be happy to open a PR, though I am not 100% sure what the good solution here is. I am thinking maybe test for an env var that can provide additional, permissible domains? (US government domain might be hardcoded with the others if someone can tell me what it is.)

While we're at it: allowing insecure connections to Azurite would be great as well - not sure that should be in the same PR though.

**Comment by ahringer** [comment]
> Hi, this really is a bit of a pain, specifically for running mockups in CI. I am with [@ahringer](https://github.com/ahringer) here, in that I run a monkeypatched version in our CI, but obviously that's not a great solution.
> 
> I'd be happy to open a PR, though I am not 100% sure what the good solution here is. I am thinking maybe test for an env var that can provide additional, permissible domains? (US government domain might be hardcoded with the others if someone can tell me what it is.)
> 
> While we're at it: allowing insecure connections to Azurite would be great as well - not sure that should be in the same PR though.

@chcharlet  - just finished PR that will finally resolve the usgovcloudapi issue. not sure which release it will be in though... I didnt do azurite as Ive never used it and seems different use case. but may look at it.
